src/pages/home/report/ReportActionItemMessageEdit.js

```diff
@@ -74,319 +74,292 @@ const defaultProps = {
-            draftMessage = parser.htmlToMarkdown(this.props.draftMessage).trim();
-        } else {
-            // We need to decode saved draft message because it's escaped before saving.
-            draftMessage = Str.htmlDecode(this.props.draftMessage);
+            return parser.htmlToMarkdown(props.draftMessage).trim();
         }
+        // We need to decode saved draft message because it's escaped before saving.
+        return Str.htmlDecode(props.draftMessage);
+    });
+    const [selection, setSelection] = useState({start: 0, end: 0});
+    const [isFocused, setIsFocused] = useState(false);
+    const [hasExceededMaxCommentLength, setHasExceededMaxCommentLength] = useState(false);
 
-        this.state = {
-            draft: draftMessage,
-            selection: {
-                start: 0,
-                end: 0,
-            },
-            isFocused: false,
-            hasExceededMaxCommentLength: false,
-        };
-    }
+    const textInputRef = useRef(null);
 
-    componentDidMount() {
+    useEffect(() => {
         // For mobile Safari, updating the selection prop on an unfocused input will cause it to automatically gain focus
         // and subsequent programmatic focus shifts (e.g., modal focus trap) to show the blue frame (:focus-visible style),
         // so we need to ensure that it is only updated after focus.
-        this.setState((prevState) => ({
-            selection: {
-                start: prevState.draft.length,
-                end: prevState.draft.length,
-            },
-        }));
-    }
-
-    componentWillUnmount() {
-        // Skip if this is not the focused message so the other edit composer stays focused.
-        if (!this.state.isFocused) {
-            return;
-        }
+        setDraft((prevDraft) => {
+            setSelection({
+                start: prevDraft.length,
+                end: prevDraft.length,
+            });
+            return prevDraft;
+        });
 
-        // Show the main composer when the focused message is deleted from another client
-        // to prevent the main composer stays hidden until we swtich to another chat.
-        ComposerActions.setShouldShowComposeInput(true);
-    }
+        return () => {
+            // Skip if this is not the focused message so the other edit composer stays focused.
+            if (!isFocused) {
+                return;
+            }
 
-    /**
-     * Update Selection on change cursor position.
-     *
-     * @param {Event} e
-     */
-    onSelectionChange(e) {
-        this.setState({selection: e.nativeEvent.selection});
-    }
+            // Show the main composer when the focused message is deleted from another client
+            // to prevent the main composer stays hidden until we swtich to another chat.
+            ComposerActions.setShouldShowComposeInput(true);
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps -- we need it to call it once on mount
+    }, []);
```

Comment:
I'm not sure about the cleanup function here. Won't `isFocused` always be `false` (as that's it's initial state).

The cleanup function is evaluated now but executed later or am I missing something?

Reply:
Unfortunately this is not working as expected. The `isFocused` method seems to return `false` where the `isFocused` state is still set to `true`. This will reintroduce this bug https://github.com/Expensify/App/issues/16623.

Reply:
https://github.com/Expensify/App/assets/16493223/344f6c2c-54f9-4f61-ba72-c25f3ae0745b

Reply:
I don't think this is the ideal approach, there could be an easy solution that we are missing. Can you please ask on Slack, given that there is many ongoing functional refactor maybe someone encountered a similar case.